java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/Server.java

```diff
@@ -885,6 +918,28 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams init) {
             );
         }
 
+        private void collectProjectCandidates(FileObject fo, List<FileObject> candidates, AtomicBoolean cancel) {
+            for (FileObject chld : fo.getChildren()) {
+                if (cancel.get()) {
+                    return;
+                }
+                if (chld.isFolder()) {
+                    Project prj = FileOwnerQuery.getOwner(chld);
+                    if (prj != null) {
+                        candidates.add(chld);
+                        Set<Project> containedProjects = ProjectUtils.getContainedProjects(prj, true);
```

Comment:
:warning:  This call is not safe for Gradle projects prior to their priming build; it may return just empty - I think the subproject structure is not read in the fallback mode. Some basic support could be perhaps added to read plain `include` directives in well known places. 
I suggest to query subprojects only after their parents prime.

Reply:
Fixed.